services/QuillLMS/app/controllers/teachers/classroom_manager_controller.rb

```diff
@@ -205,6 +214,24 @@ def activity_feed
     render json: { data: data_for_activity_feed(current_user) }
   end
 
+  private def generate_onboarding_checklist
+    Objective::ONBOARDING_CHECKLIST_NAMES.map do |name|
+      objective = Objective.find_by_name(name)
+      checkbox = Checkbox.find_by(objective: objective, user: current_user)
+
+      # handles case where user has been using Quill since before we introduced the new objectives
+      if objective && !checkbox && [Objective::EXPLORE_OUR_LIBRARY, Objective::EXPLORE_OUR_DIAGNOSTICS].include?(name) && current_user.units&.any?
```

Comment:
A couple of notes on the conditional:
1.  When would `objective` be nil?  (Or maybe a different question is when is `build_objectives.rake` called)
2.  Maybe extract `[Objective::EXPLORE_OUR_LIBRARY, Objective::EXPLORE_OUR_DIAGNOSTICS].include?(name)` into a class method on `Objective`  (e.g.  

```
def self.newer_objectives?(name)
   [EXPLORE_OUR_LIBRARY, EXPLORE_OUR_DIAGNOSTICS].include?(name)
end
```

Reply:
1.`objective` might be nil in someone's local environment, since we only run `build_objectives.rake` during the initial install, or if there is a window between when this code gets deployed or when the new objectives are created (we want to create them as close as possible to when the code is live, because otherwise the existing UI will get weird because there are places on the frontend we pull all the objectives at once). I have a mini script I will run when this gets deployed, but it seems safer to include a nil check.

2. I thought about that, but I just feel like eventually `newer_objectives` will be an outdated name and it will just be confusing. I feel like it's clearer to just have this comment where the code actually gets used.